src/plugins/intel_cpu/src/nodes/common/reorder_prim.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <ie_common.h>
+#include <node.h>
+
+#include <memory>
+
+namespace ov {
+namespace intel_cpu {
+
+std::shared_ptr<dnnl::primitive> getReorderPrim(MultiCachePtr cache,
+                                                const dnnl::engine& engine,
+                                                const dnnl::memory::desc& src,
+                                                const dnnl::memory::desc& dest,
+                                                impl_desc_type* p_impl_type = nullptr);
+
+inline std::shared_ptr<dnnl::primitive> getReorderPrim(MultiCachePtr cache,
```

Comment:
why do we need overload getReorderPrim instead of keeping just one?

Reply:
Seems like we don't have always access to `engine` to set it as parameter

Reply:
Will remove extra overloads to simplify APIs